Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1

```diff
@@ -0,0 +1,408 @@
+
+    <#
+    .DESCRIPTION
+
+    Specifies a configured CPU reservation in MHz
+    #>
+    [DscProperty()]
+    [long] $CpuReservationMHz = 0
+
+    <#
+    .DESCRIPTION
+
+    Specifies CPU shares level. The valid values are Custom, High, Low, Normal and Unset.
+    #>
+    [DscProperty()]
+    [SharesLevel] $CpuSharesLevel = [SharesLevel]::Normal
+
+    <#
+    .DESCRIPTION
+
+    Indicates that Memory expandable reservation is enabled
+    #>
+    [DscProperty()]
+    [bool] $MemExpandableReservation = $true
+
+    <#
+    .DESCRIPTION
+
+    Specifies a configured memory limit in GB
+    #>
+    [DscProperty()]
+    [long] $MemLimitGB = -1
+
+    <#
+    .DESCRIPTION
+
+    Specifies a configured memory reservation in GB
+    #>
+    [DscProperty()]
+    [long] $MemReservationGB = 0
+
+    <#
+    .DESCRIPTION
+
+    Specifies Memory shares level. The valid values are Custom, High, Low, Normal and Unset.
+    #>
+    [DscProperty()]
+    [SharesLevel] $MemSharesLevel = [SharesLevel]::Normal
+
+    <#
+    .DESCRIPTION
+
+    Specifies the memory allocation level for the resource pool.
+    This parameter is ignored unless MemSharesLevel is set to Custom.
+    #>
+    [DscProperty()]
+    [long] $NumMemShares
+
+    <#
+    .DESCRIPTION
+
+    Specifies the CPU allocation level for the resource pool.
+    This DSC property is ignored unless CpuSharesLevel is set to Custom.
+    #>
+    [DscProperty()]
+    [long] $NumCpuShares
+
+    <#
+    .DESCRIPTION
+
+    Specifies whether the DRS rule should be present or absent.
+    #>
+    [DscProperty(Mandatory)]
+    [Ensure] $Ensure
+
+    <#
+    .DESCRIPTION
+
+    Specifies the instance of the 'InventoryUtil' class that is used
+    for Inventory operations.
+    #>
+    hidden [InventoryUtil] $InventoryUtil
+
+    hidden [string] $CpuExpandableReservationParameterName = 'CpuExpandableReservation'
+    hidden [string] $CpuLimitMHzParameterName = 'CpuLimitMHz'
+    hidden [string] $CpuReservationMHzParameterName = 'CpuReservationMHz'
+    hidden [string] $CpuSharesLevelParameterName = 'CpuSharesLevel'
+    hidden [string] $MemExpandableReservationParameterName = 'MemExpandableReservation'
+    hidden [string] $MemLimitGBParameterName = 'MemLimitGB'
+    hidden [string] $MemReservationGBParameterName = 'MemReservationGB'
+    hidden [string] $MemSharesLevelParameterName = 'MemSharesLevel'
+    hidden [string] $NumCpuSharesParameterName = 'NumCpuShares'
+    hidden [string] $NumMemSharesParameterName = 'NumMemShares'
+    hidden [hashtable] $CpuSharesMap = @{
+       'Low' = 2000
+       'Normal' = 4000
+       'High' = 8000
+    }
+    hidden [hashtable] $MemSharesMap = @{
+       'Low' = 81920
+       'Normal' = 163840
+       'High' = 327680
+    }
+
+    [void] Set() {
+        try {
+            $this.ConnectVIServer()
+            $this.InitInventoryUtil()
+            $resourcePool = $this.InventoryUtil.GetResourcePool($this.ResourcePoolName, $this.ResourcePoolLocation)
```

Comment:
You can change the **`GetResourcePool`** method signature to accept the parent directly. This way you don't have to call the **`GetResourcePoolParent`** more than once per method. Currently if you create a new **`ResourcePool`**, you call it twice, once to check if it exists and another when you're creating it.

Reply:
Fixed